src/main/java/io/cdap/plugin/gcp/publisher/source/PubSubSubscriberUtil.java

```diff
@@ -86,4 +115,91 @@ protected static JavaDStream<PubSubMessage> getInputDStream(StreamingContext str
     return new JavaDStream<>(dStream, tag);
   }
 
+  /**
+   * Create a new subscription (if needed) for the supplied topic.
+   *
+   * @param preCheck      Any checks that need to be applied before each retry.
+   * @param backoffConfig {@link BackoffConfig} for retries.
+   * @param subscription  Subscription name string.
+   * @param topic         Topic name string.
+   * @param clientSupplier Supplier for creating {@link SubscriptionAdminClient}
+   * @param isRetryableException Predicate for checking if the exception is retryable.
+   * @throws InterruptedException If the wait for retry is interrupted.
+   * @throws IOException          If {@link SubscriptionAdminClient} cannot be created.
+   */
+  public static void createSubscription(BooleanSupplier preCheck, BackoffConfig backoffConfig, String subscription,
```

Comment:
This preCheck looks like a `Predicate`. signatures are similar, but `Predicate` is sematically better suited here.

Reply:
Predicate takes an argument, in this case the function used in original code did not need one.